### ISSUE TITLE
[Formatter] Slash of last `//` comment of indented code deleted

### ISSUE THREAD

**Issue by JohelEGP** [llvm/llvm-project#53799]
`.clang-format`:
```yaml
SpacesInLineCommentPrefix:
  Minimum: 1
  Maximum: 1
```
Input:
```C++
//   x
// y
```
Expected output:
```C++
// x
// y
```
Actual output:
```C++
// x
/y
```

I noticed this a while back, presumably after the introduction of `SpacesInLineCommentPrefix`.

**Comment by mkurdej** [comment]
Interesting, the more spaces you have before `x`, the more characters are eaten before `y`...
```
//     x
// y
```
becomes
```
// xy
```

**Comment by mkurdej** [comment]
Ha! If anyone had still any doubt that integer promotion rules weren't a difficult topic...
![image](https://user-images.githubusercontent.com/1609564/153775535-f9fcc2a4-ebef-4727-942e-75296f64303f.png)

**Comment by mkurdej** [comment]
Revision URI: https://reviews.llvm.org/D119680

**Comment by JohelEGP** [comment]
Excellent, thanks!

**Comment by Sam0523** [comment]
When will the fix be applied in a release version? I found this issue still existing in 14.x.

Wouldn't the fix be backported to 14.x? I assumed this is a severe bug.

**Comment by tstellar** [comment]
@HazardyKnusperkeks What do you think about backporting this? https://github.com/llvm/llvm-project/commit/e967d97a35a93c883528a9672159edff05f5addb

**Comment by tstellar** [comment]
I've reviewed this and it seems OK to me.

**Comment by tstellar** [comment]
Merged: d0cd5a872f8d0d3720d2428dd420ae9177e6a8f1
